contracts/test/utils.ts

```diff
@@ -50,9 +55,42 @@ export async function setNextBlockTimestamp(timestamp: BigNumberish): Promise<nu
   return timestamp;
 }
 
+// --- Uniswap SDK Helpers ---
+export async function encodeRoute(tokens: SupportedToken[]): Promise<string> {
+  const route = await getRoute(tokens);
+  return encodeRouteToPath(route, false);
+}
+
+async function getRoute(tokens: SupportedToken[]): Promise<Route<Token, Token>> {
+  const poolPromises: Promise<Pool>[] = [];
+  tokens.forEach((_, index) => {
+    if (index > 0) poolPromises.push(getPoolInstance(tokens[index - 1], tokens[index]));
+  });
+  const pools = await Promise.all(poolPromises);
+  return new Route(pools, pools[0].token0, pools[pools.length - 1].token1);
+}
+
+async function getPoolInstance(token0: SupportedToken, token1: SupportedToken): Promise<Pool> {
+  // Get pool contract
+  token0 = token0 === 'eth' ? 'weth' : token0;
+  token1 = token1 === 'eth' ? 'weth' : token1;
+  const tokenA = new Token(1, tokens[token0].address, tokens[token0].decimals, tokens[token0].symbol, tokens[token0].name); // prettier-ignore
+  const tokenB = new Token(1, tokens[token1].address, tokens[token1].decimals, tokens[token1].symbol, tokens[token1].name); // prettier-ignore
+  const fee = FeeAmount.MEDIUM; // always use Medium fee for now
+  const poolAddress = computePoolAddress({ factoryAddress: UNISWAP_FACTORY, tokenA, tokenB, fee });
```

Comment:
more of a question! 
looking at https://github.com/Uniswap/uniswap-v3-sdk/blob/cc9acb1d27cf6c7f02be167f8747e4d0e9c99c7c/src/utils/computePoolAddress.ts#L20 
the assumption here is that tokenA, tokenB LP exists ?  It doesn't account for a situation where it does not right ?

Reply:
Right, we don't verify the pool exists, and if the pool doesn't exist the swap will fail. This is ok for now but when integrated into the frontend we'd want to verify the pool exists